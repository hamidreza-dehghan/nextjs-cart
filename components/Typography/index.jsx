/** @jsxImportSource @emotion/react */
import React from 'react'
import styled from '@emotion/styled'
import { css } from '@emotion/react'
import theme from 'theme'

const getStyles = ({ variant }) => css`
  font-family: ${theme.typography[variant]?.fontFamily};
  font-size: ${theme.typography[variant]?.fontSize}em;
  font-weight: ${theme.typography[variant]?.fontWeight};
  margin: ${theme.typography[variant]?.margin};
  color: ${theme.typography[variant]?.color};
  text-decoration: ${theme.typography[variant]?.textDecoration};
`
const StyledTypography = styled.div`
  ${(variant) => getStyles(variant)}
`
export function Typography({ variant, children, ...props }) {
  return (
    <StyledTypography as={variant} variant={variant} {...props}>
      {children}
    </StyledTypography>
  )
}

export default Typography
