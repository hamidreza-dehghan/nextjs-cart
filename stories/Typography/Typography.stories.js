import { Typography } from 'components/Typography'

export default {
  title: 'Ultimate/Components/Typography',
  component: Typography,
  tags: ['autodocs'],
  argTypes: {
    variant: 'h1',
    children: 'Heading 1',
  },
}

export const H1 = {
  args: {
    variant: 'h1',
    children: 'Heading 1',
  },
}

export const H2 = {
  args: {
    variant: 'h2',
    children: 'Heading 2',
  },
}

export const H3 = {
  args: {
    variant: 'h3',
    children: 'Heading 3',
  },
}

export const H4 = {
  args: {
    variant: 'h4',
    children: 'Heading 4',
  },
}

export const H5 = {
  args: {
    variant: 'h5',
    children: 'Heading 5',
  },
}

export const H6 = {
  args: {
    variant: 'h6',
    children: 'Heading 6',
  },
}

export const Paragraph = {
  args: {
    variant: 'p',
    children: 'Paragraph',
  },
}

export const Subtitle = {
  args: {
    variant: 'sub',
    children: 'Subtitle',
  },
}

export const Pre = {
  args: {
    variant: 'pre',
    children: 'The code comes here...',
  },
}

export const Link = {
  args: {
    variant: 'a',
    href: '#',
    children: 'Link',
  },
}
