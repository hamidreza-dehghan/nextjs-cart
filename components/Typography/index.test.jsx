import React from 'react'
import renderer from 'react-test-renderer'
import Typography from '.'

describe('Typography', () => {
  test('Renders correctly', () => {
    const component = renderer.create(<Typography />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('H1', () => {
    const component = renderer.create(
      <Typography variant="H1">Hello World</Typography>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('H2', () => {
    const component = renderer.create(
      <Typography variant="H2">Hello World</Typography>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('H3', () => {
    const component = renderer.create(
      <Typography variant="H3">Hello World</Typography>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('H4', () => {
    const component = renderer.create(
      <Typography variant="H4">Hello World</Typography>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('H5', () => {
    const component = renderer.create(
      <Typography variant="H5">Hello World</Typography>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('H6', () => {
    const component = renderer.create(
      <Typography variant="H6">Hello World</Typography>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
