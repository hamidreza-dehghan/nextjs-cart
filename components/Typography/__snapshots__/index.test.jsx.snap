// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography H1 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-n9tuf1-StyledTypography-getStyles{font-size:em;}",
      }
    }
    data-emotion="css n9tuf1-StyledTypography-getStyles"
  />,
  <H1
    className="css-n9tuf1-StyledTypography-getStyles e1imt8qb0"
    variant="H1"
  >
    Hello World
  </H1>,
]
`;

exports[`Typography H2 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-n9tuf1-StyledTypography-getStyles{font-size:em;}",
      }
    }
    data-emotion="css n9tuf1-StyledTypography-getStyles"
  />,
  <H2
    className="css-n9tuf1-StyledTypography-getStyles e1imt8qb0"
    variant="H2"
  >
    Hello World
  </H2>,
]
`;

exports[`Typography H3 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-n9tuf1-StyledTypography-getStyles{font-size:em;}",
      }
    }
    data-emotion="css n9tuf1-StyledTypography-getStyles"
  />,
  <H3
    className="css-n9tuf1-StyledTypography-getStyles e1imt8qb0"
    variant="H3"
  >
    Hello World
  </H3>,
]
`;

exports[`Typography H4 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-n9tuf1-StyledTypography-getStyles{font-size:em;}",
      }
    }
    data-emotion="css n9tuf1-StyledTypography-getStyles"
  />,
  <H4
    className="css-n9tuf1-StyledTypography-getStyles e1imt8qb0"
    variant="H4"
  >
    Hello World
  </H4>,
]
`;

exports[`Typography H5 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-n9tuf1-StyledTypography-getStyles{font-size:em;}",
      }
    }
    data-emotion="css n9tuf1-StyledTypography-getStyles"
  />,
  <H5
    className="css-n9tuf1-StyledTypography-getStyles e1imt8qb0"
    variant="H5"
  >
    Hello World
  </H5>,
]
`;

exports[`Typography H6 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-n9tuf1-StyledTypography-getStyles{font-size:em;}",
      }
    }
    data-emotion="css n9tuf1-StyledTypography-getStyles"
  />,
  <H6
    className="css-n9tuf1-StyledTypography-getStyles e1imt8qb0"
    variant="H6"
  >
    Hello World
  </H6>,
]
`;

exports[`Typography Renders correctly 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-n9tuf1-StyledTypography-getStyles{font-size:em;}",
      }
    }
    data-emotion="css n9tuf1-StyledTypography-getStyles"
  />,
  <div
    className="css-n9tuf1-StyledTypography-getStyles e1imt8qb0"
  />,
]
`;
